### PR TITLE
*: add missing colon to errorf messages to improve readability

### PR DESCRIPTION
**Pull request by ash2k**
Mostly add missing colons but also address few other minor issues.

It's confusing to read an error message without a colon for nesting, so that was the main motivation for the change. Happened to me couple of times and I couldn't find the whole string anywhere in code, wasted some time on it.

RELEASE NOTES:
* none

### CODE REVIEW

**Comment by ash2k** [comment]
Tests pass locally for me, I think it's a flaky test. Cannot rerun the action - no permissions.

File: interop/xds/server/server.go [hunk 1]
```diff
@@ -118,9 +118,10 @@ func main() {
 	// If -secure_mode is not set, expose all services on -port with a regular
 	// gRPC server.
 	if !*secureMode {
-		lis, err := net.Listen("tcp4", fmt.Sprintf(":%d", *port))
+		addr := fmt.Sprintf(":%d", *port)
```
Comment (arvindbr8) [inline interop/xds/server/server.go:121]: nit: maybe init `addr` above this block and nix line 145?

**Review by arvindbr8 — APPROVED** [review]
lgtm, pinging @easwars for another set of eyes

File: interop/xds/server/server.go (hunk 1 above)
Comment (ash2k) [reply interop/xds/server/server.go:121]: I wanted to preserve the "laziness" of formatting, if that's ok.

**Comment by ash2k** [comment]
That's what I did - regex, but also checked some function usage manually. I certainly may have missed some places - there is a lot of code.

File: xds/internal/resolver/xds_resolver.go [hunk 2]
```diff
@@ -237,7 +237,7 @@ func (r *xdsResolver) run() {
 			return
 		case update := <-r.updateCh:
 			if update.err != nil {
-				r.logger.Warningf("Watch error on resource %v from xds-client %p, %v", r.ldsResourceName, r.client, update.err)
+				r.logger.Warningf("Watch error on resource %v from xds-client %p: %v", r.ldsResourceName, r.client, update.err)
```
Comment (easwars) [inline xds/internal/resolver/xds_resolver.go:240]: I think we can skip this. I'm actually working on another PR where I'm getting rid of this log message (along with a few others) as it is redundant.

File: xds/internal/xdsclient/xdsresource/unmarshal_cds.go [hunk 3]
```diff
@@ -503,7 +503,7 @@ func outlierConfigFromCluster(cluster *v3clusterpb.Cluster) (*OutlierDetection,
 	interval := defaultInterval
 	if i := od.GetInterval(); i != nil {
 		if err := i.CheckValid(); err != nil {
-			return nil, fmt.Errorf("outlier_detection.interval is invalid with error %v", err)
+			return nil, fmt.Errorf("outlier_detection.interval is invalid with error: %v", err)
```
Comment (easwars) [inline xds/internal/xdsclient/xdsresource/unmarshal_cds.go:506]: Can we skip the `with error` part in these error strings:
`return nil, fmt.Errorf("outlier_detection.interval is invalid: %v", err)`

**Review by easwars — CHANGES_REQUESTED** [review]
Some minor nits. Thank you so much for doing this.

**Review by easwars — APPROVED** [review]

**Comment by easwars** [comment]
Thank you very much for your contribution !!

**Pull request merged**